core/src/main/java/nu/yona/server/messaging/entities/Message.java

```diff
@@ -199,4 +220,34 @@ public void prepareForDelete()
 	protected abstract void encrypt();
 
 	protected abstract void decrypt();
+
+	/**
+	 * Creates a duplicate of the given messages, ready to be sent to a destination. Everything is identical to the original one,
+	 * except for:
+	 * <ul>
+	 * <li>The ID, which will be 0</li>
+	 * <li>References to other messages (e.g. replies). If this is ever necessary, it needs to be designed and built</li>
+	 * <li>Reference to the message destination. The duplicate is ready to be sent and does not have a destination yet.</li>
+	 * </ul>
+	 * For this to work, the message class needs to have a public copy constructor.
```

Comment:
This currently has been only defined for possible direct messages, right? Maybe good to document that here.

Reply:
Makes sense.